libraries/testing/include/eosio/testing/snapshot_suites.hpp

```diff
@@ -106,5 +106,68 @@ struct buffered_snapshot_suite {
    }
 };
 
-using snapshot_suites = boost::mpl::list<variant_snapshot_suite, buffered_snapshot_suite>;
+
+struct json_snapshot_suite {
+   using writer_t = ostream_json_snapshot_writer;
+   using reader_t = istream_json_snapshot_reader;
+   using write_storage_t = std::ostringstream;
+   using snapshot_t = std::string;
+   using read_storage_t = std::istringstream;
+
+   struct writer : public writer_t {
+      writer( const std::shared_ptr<write_storage_t>& storage )
+      :writer_t(*storage)
+      ,storage(storage)
+      {
+
+      }
+
+      std::shared_ptr<write_storage_t> storage;
+   };
+
+   static std::string temp_file() {
+      static fc::temp_directory temp_dir;
```

Comment:
Don't use a `static` here. Instead have the reader create its own `temp_directory` and then its destructor can remove it. Using a `static` here means that every call to `temp_file()` returns the same file, which seems counterintuitive.

Reply:
It has to be used here and not within the reader struct because it is needed in the static function `get_reader` which constructs the reader. This code is also only relevant in the `snapshot tests`.

Reply:
I think it is worth cleaning up. Probably need to refactor the `get_reader` (just remove and use types directly).

Reply:
Sorry, I complete misread how this was being used.